src/content/1.7/development/components/grid-component.md

```diff
@@ -7,229 +7,404 @@ weight: 2
 
-## Grid Data Factory
+We have just created a basic Grid Definition factory in which we defined our Grid's id `products`, translatable
+name `Products` and 3 data columns.
+
+{{% notice note %}}
+It is _recommended_ to keep your name translatable. To make that easy
+`AbstractGridDefinitionFactory` provides access to translator via `trans()` method.
+{{% /notice %}}
+
+And finally register your Grid definition factory as a service.
+
+```yaml
+prestashop.core.grid.definition.factory.product_grid_definition_factory:
+    class: 'PrestaShop\PrestaShop\Core\Grid\Definition\Factory\ProductGridDefinitionFactory'
+    parent: 'prestashop.core.grid.definition.factory.abstract_grid_definition'
+    public: true
+``` 
+
+Most of the time you won't be creating Grid Definition by yourself but delegating this task to other services.
+But in case you need to create Grid Definition by hand, here's how you can do that.
+
+```php
+$productsGridDefinitionFactory = $container->get('prestashop.core.grid.definition.factory.product_grid_definition_factory');
+$productsGridDefinition = $productsGridDefinitionFactory->getDefinition();
 
-As you can imagine, the responsibility of the Grid Data Factory is to retrieve the Grid data, based on the Grid Definition and the Grid Query Builder.
+// you can access all information thats was defined
+$productsGridDefinition->getColumns(); // collection of defined columns
+$productsGridDefinition->getName(); // "Products"
+$productsGridDefinition->getId(); // "products"
+``` 
 
-The only method available of `GridDataFactoryInterface` is `getData` which returns an instance of `GridData`.
-A GridData is an **immutable object** used to store and retrieve the grid data, so if you want to alter this data, you must do it in Grid Data Factory *before* the GridData creation.
+## Search Criteria
 
-You can rely on the implementation provided in the component `GridDataFactory` or create your owns if you have specific needs. 
+In Grid component Search Criteria is used for Grid's data sorting, paginating & filtering. Search Criteria
+can be loaded from database, URL query or anywhere else. 
 
-## Grid Query Builder
+{{% notice info %}}
+Grid component itself does not manage Search Criteria but instead it provides interface for it.
+In PrestaShop [Filters component](#) is used to resolve Search Criteria for Grid.
+{{% /notice %}}
 
-The Grid Query Builder is responsible of building the right query to retrieve the data according to the Grid definition and Search filters that come from the User request.
+{{% notice note %}}
+Search Criteria is immutable. This means that once Search Criteria is created it cannot be changed.
+{{% /notice %}}
 
-This is the related interface:
+### Creating Search Criteria
+
+Even though most of the time Search Criteria will be created using [Filters component](#), you can still
+create it by yourself. Grid provides simple implementation for it.
+
+```php
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteria;
```

Comment:
weird thing, why is `SearchCriteriaInterface` the only one in `PrestaShop\PrestaShop\Core\Grid\Search` namespace when all filters that implement it are in the `PrestaShop\PrestaShop\Core\Search` namespace, shoudn't the Interface move?
either it or the Filters classes

Reply:
ok I get the idea But then if Filters are only used by the Grid, which would appear so since they implement the `SearchCriteriaInterface ` shouldn't they be in the Grid namespace as well